### ISSUE TITLE
Slicing after `map_overlap` with asymmetric overlap gives unexpected result.

### ISSUE THREAD

**Issue by benjeffery** [dask/dask#5448]
I'm seeing unexpected results from slicing after a `map_overlap` with an asymmetric overlap. For example:  (dask  == 2.5.0)

```
import dask.array as da
import numpy as np
arange = da.from_array(np.arange(20), chunks=(10,))
after_overlap = a.map_overlap(lambda x:x, depth={0:(0,5)}, boundary="none", trim=True)
print(after_overlap.compute())

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]
```
As expected. 
We then try to slice:
```
print(after_overlap[1:].compute())

[ 1  2  3  4 10 11 12 13 14 15 16 17 18 19]
```
There is a missing block of numbers (the same size as the overlap) in the middle.
Slicing from the other end:
```
print(after_overlap[:-1].compute())

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]
```
Seems to have no effect. If you use `depth=5` everything works as expected.

I will start to look into this, but thought I would write up first incase anyone has an idea.

Thanks for dask, been loving using it!

**Comment by benjeffery** [comment]
I think I have found the cause.
In https://github.com/dask/dask/blob/master/dask/array/overlap.py#L160 there is the following:
```
            left = [bds[0] + left_depth]
            right = [bds[-1] + right_depth]
```
This code is calculating the size of chunks at the edges of the array, so we need to add the chunk size to the boundary size, BUT, the left edge chunk has a _right_ size overlap, not a _left_ size overlap, so swapping these would seem to be correct:
```
            left = [bds[0] + right_depth]
            right = [bds[-1] + left_depth]
```
I'm pretty sure this is correct, but it is possible I have misunderstood the intent.
I'll write up a PR for this.

**Comment by quasiben** [comment]
I am seeing the same thing.  Thanks for the report and the fix!

**Comment by mrocklin** [comment]
Fixed by #5449
